.github/workflows/run-tests.yml

```diff
@@ -508,27 +512,29 @@ jobs:
             persistence_driver: postgres12_pgx
             parallel_flags: "-parallel=2" # reduce parallelism for postgres
             containers: [postgresql]
-    runs-on: ubuntu-20.04
+    runs-on: ${{ fromJson(needs.set-up-single-test.outputs.runs_on) }}
     env:
       PERSISTENCE_TYPE: ${{ matrix.persistence_type }}
       PERSISTENCE_DRIVER: ${{ matrix.persistence_driver }}
-      BUILDKITE_MESSAGE: '{"job": "functional-test-xdc", "db": "${{ matrix.persistence_driver }}"}'
+      TEST_TIMEOUT: ${{ needs.set-up-single-test.outputs.test_timeout }}
```

Comment:
nit: Set this in `Run functional test xdc` so it's clear that it doesn't apply to the flaky test detection step.